packages/sdk-utilities/src/lib/authorize.utils.ts

```diff
@@ -1,33 +1,55 @@
+/**
+ *
+ * Copyright © 2025 Ping Identity Corporation
+ *
+ **/
+
 /**
  * Import the PKCE and ResponseType utilities from the JavaScript SDK
  */
-import { PKCE, ResponseType } from '@forgerock/javascript-sdk';
-import { generateAndStoreAuthUrlValues } from '@forgerock/javascript-sdk/src/oauth2-client/state-pkce';
+import PKCE from './pkce.utils.js';
+import {
+  CreateAuthorizeUrlOptions,
+  GenerateAndStoreAuthUrlValues,
+  ResponseType,
+} from './authorize.utils.types.js';
+
+function getStorageKey(clientId: string, prefix?: string) {
+  return `${prefix || 'FR-SDK'}-authflow-${clientId}`;
+}
 
 /**
- * Define the options for the authorization URL
- * @param clientId The client ID of the application
- * @param redirectUri The redirect URI of the application
- * @param responseType The response type of the authorization request
- * @param scope The scope of the authorization request
+ * Generate and store PKCE values for later use
+ * @param { string } storageKey - Key to store authorization options in sessionStorage
+ * @param {GenerateAndStoreAuthUrlValues} options - Options for generating PKCE values
+ * @returns { state: string, verifier: string, GetAuthorizationUrlOptions }
  */
-export interface GetAuthorizationUrlOptions {
-  clientId: string;
-  login: 'redirect';
-  redirectUri: string;
-  responseType: string;
-  scope: string;
+function generateAndStoreAuthUrlValues(options: GenerateAndStoreAuthUrlValues) {
```

Comment:
This can't be a utility as it would violate the "utilities are side-effect free, stateless functions". The fact that this interacts with `sessionStorage` would mean if would have to "live" within the Effects package.